Assets/Scripts/TriggeredTorch.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TriggeredTorch : MonoBehaviour
+{
+ 
+   [SerializeField]
+    private GameObject Torch;
+    private Collider2D fireBall;
+    private Collider2D onTorchCol;
+    public Sprite onTorch; 
+
+     void Start()
+    {
+        fireBall = GameObject.Find("Fireball").GetComponent<Collider2D>();
```

Comment:
This will only find the first fireball in the scene and will not work for multiple fireballs nor fireballs that are spawned/instantiated during gameplay. But since im going to have you use triggers instead, you'll just be able to get rid of this. If you wanted a reference to the actual fireball that collided with the torch, you can actually use the "collision" variable thats passed through in the OnCollisionEnter2D. This also applies to OnTriggerEnter2D